Feat/notice

Aperture507 opened the pull request:
회원 목록 | 공지사항
--------|---------
<img src="https://user-images.githubusercontent.com/81297398/126593017-2b90e1fb-5126-4372-a57e-30caef2905ca.png"  > | <img src="https://user-images.githubusercontent.com/81297398/126593034-fff66740-affd-4688-827f-21ace66cf11c.png"  >

- 아직 FlatList에 버튼 기능 x
- 회원 목록의 검색 창에도 검색 기능 x
  - 회원 목록 창에서 뜨는 Warning은 keyExtractor를 studentCode로 받아오는데, 그 StudentCode가 현재 임의로 지정된 중복된 숫자들이기 때문에 발생한다. => **데이터를 받아오게 된다면 없어질 Warning**

### 디자인적인 부분만 완성됐습니다!     버튼, 네비게이트 등은 추후 추가해야합니다~~

DonghyunKim98 commented:
- 지금 lint fail 이 나오고 있는데 확인해주세요.
- `yarn lint-fix` 를 터미널에 입력하면 자동으로 린트를 고칠 수 있는 것들은 고치고 못 고치는 부분들 확인할 수 있을겁니다. (린트를 제대로 설정했다면)
- 위의 명령어가 동작하지 않는다면, 클라이언트 `package.json` 을 참고해서 동작하게 바꾸어놔주세요.

DonghyunKim98 commented:
> **체크해야할 점**
> 
> * 제플린 참고하여 스타일에 비율을 그대로 적용시킬 수 있는 방법?
> * '카카오톡으로 로그인' 버튼 기능만 있고 실제 로그인 구현 필요

- 비율을 그대로 적용시킬 수 있는 방법,,, 일단 재헌이가 알려준데로 option 키를 눌러서 %를 이용하니깐 거의 비슷하긴 한데, 이게 제대로 먹히는 건지는 의문이 드는것이 사실
- 엄청 섬세한 퍼블리싱은 어떻게 하는건지 이번 플젝에서 나도 더 배워갔으면 좋겠다 😅

Aperture507 committed:
feat: 공지사항 검색창 추가

Aperture507 committed:
fix: lint 해결

Aperture507 committed:
feat: 역할 색깔 표시 추가

Aperture507 committed:
feat: 무지성 FlatList

Aperture507 committed:
fix: Member Notice 바뀐거 수정

DonghyunKim98 commented on app/components/screens/notice/Notice.tsx:
```diff
@@ -26,14 +55,24 @@ function Notice() {
+            placeholderTextColor="#a2a2a2"
+          />
+          <Image
+            source={require('../../../assets/images/searchIcon.png')}
```
- asset 은 위로 따로 빼주세요.

DonghyunKim98 commented on app/components/screens/notice/Notice.tsx:
```diff
@@ -10,11 +10,40 @@ import {faChevronLeft} from '@fortawesome/free-solid-svg-icons';
+  },
+  SectionStyle: {
+    flexDirection: 'row',
+  },
```
- Object 의 key 값은 대문자로 보통 시작하지 않습니다.
- `style` 이라는 부분이 겹칩니다. 제외시켜주세요.

DonghyunKim98 commented on app/components/screens/notice/Notice.tsx:
```diff
@@ -43,6 +43,44 @@ const styles = StyleSheet.create({
+    color: '#00203f',
+    marginTop: heightPercentage(-4),
+    marginLeft: widthPercentage(6),
+  },
```
- 객체 안의 key 값은 대문자로 시작하지 않습니다!

DonghyunKim98 commented on app/components/screens/notice/Notice.tsx:
```diff
@@ -43,6 +43,44 @@ const styles = StyleSheet.create({
+    fontFamily: 'NotoSansKR-Regular',
+    fontSize: 12,
+    color: '#00203f',
+    marginTop: heightPercentage(-4),
```
- 제대로 구동은 안해봐서 모르는데, - 값을 주어야 퍼블리싱이 나오나요?

Aperture507 replied:
Member.tsx 의 37번째 줄의 imageStyle 값에서도 
marginLeft: widthPercentage(-40)
처럼 -값을 줬는데, 검색창이랑 검색 돋보기 아이콘을 flexDirection: 'row' 로 묶어서 넣으니까 검색창 뒤에 돋보기 아이콘이 로드됩니다. 
![Simulator Screen Shot - iPhone 12 - 2021-07-21 at 14 06 13](https://user-images.githubusercontent.com/81297398/126434005-0dd455ba-570e-4ef6-8a98-557a229bd47b.png)

근데 원하는건 검색창 안에 돋보기 아이콘이 있는 것이니까 그럼 TextInput 안에 Image를 넣으면 되나..? 라고 생각했는데 그런 방식은 또 불가능하고 그래서 일단은 -40값을 주어서 왼쪽으로 움직여서 억지로 겹치게 한 느낌입니다. 

저는 이 돋보기 모양을 터치해야 검색되는게 아니고 검색어를 넣자마자 바로바로 결과값이 로드되는 형식으로 일단 생각하고 있어서 단순 장식으로만 생각하고 있는지라 이렇게 대충 위치만 맞게 넣었는데, 이 부분 관련해서도 논의가 필요할 것 같습니다.


pr 주신 -4값 가진 것도 단순 표시이기 때문에 큰 상관 없을 것 같아서 그렇게 했습니다

(marginTop: heightPercentage(0), 준 화면)
![Simulator Screen Shot - iPhone 12 - 2021-07-21 at 14 10 53](https://user-images.githubusercontent.com/81297398/126434365-c3f8b0e8-8bc1-4b06-be93-9c2e68f9e1b5.png)

DonghyunKim98 replied:
### 상단 TextInput 관련
```tsx
<View>
   <TextInput/>
   <FontAwesomeIcon>
</View>
```
이런식으로 짜면 됩니다.

Aperture507 replied:
어... FontAwesomeIcon으로 로컬 이미지 파일을 불러오는 것도 가능한건가요?

DonghyunKim98 replied:
아 로컬 이미지 파일이야? 너가 직접 찾은?
그러면 `Image` 로 진행하면 되지!

Aperture507 replied:
성진님이 올려주신 PNG 파일 씁니다 
```tsx
        <View style={styles.searchSection}>
          <TextInput
            style={styles.searchTextInput}
            placeholder="검색"
            placeholderTextColor="#a2a2a2"
          />
          <Image source={searchIcon} style={styles.imageStyle} />
        </View>
```
현재 코드 형태랑 무언가 다른건가요?

DonghyunKim98 replied:
```tsx
searchSection: {
    width: '90%'
    height: heightPercentage(58),
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    borderRadius: widthPercentage(10),
    backgroundColor: 'white',
  },
  searchTextInput: {
    width: '85%',
    height: '80%',
    fontSize: fontPercentage(20),
    lineHeight: fontPercentage(25),
    fontFamily: 'NotoSansKR-Bold',
    paddingLeft: widthPercentage(13),
  },
  imageStyle: {
    width: widthPercentage(25),
    height: heightPercentage(25),
    resizeMode: 'contain',
  },
```

DonghyunKim98 replied:
이런식으로 퍼블리싱 할 것 같습니다. 참고해주세요

DonghyunKim98 replied:
부모에다가 `justifyContent` 랑 `alignItems` 를 걸어야합니다~

DonghyunKim98 replied:
```tsx
roleSection: {
    flexDirection: 'row',
    marginTop: heightPercentage(15),
    marginLeft: widthPercentage(23),
    alignItems: 'center', // 추가됨
  },
  ```
  이렇게 부모에 정렬 걸어주면 됩니다.

DonghyunKim98 commented on app/components/screens/notice/Notice.tsx:
```diff
@@ -43,6 +43,44 @@ const styles = StyleSheet.create({
+  },
+  roleTextStyle: {
+    fontFamily: 'NotoSansKR-Regular',
+    fontSize: 12,
```
- 반응형 작업해주세요 😅

DonghyunKim98 commented:
### 7.20 오후 10시 13분
- 크게 공통적으로 보이는 부분은 3가지 입니다.
    1. 반응형 작업이 덜 되어 있다.
    2. 네이밍 규칙을 지키지 못하고 있다.
    3. type 을 명시하지 못하고 있다.
- 위 3가지만 먼저 해결하시면 될 것 같습니다. `FlatList` 관련 Typescript 코드는 조만간 또 공유해드리겠습니다!

Aperture507 committed:
fix: asset 위치 수정

Aperture507 committed:
fix: Object key 값 소문자로 수정

Aperture507 committed:
fix: type 명시

Aperture507 committed:
fix: 반응형 작업

Aperture507 committed:
fix: 오타 수정

Aperture507 committed:
fix: 검색 아이콘 코드 수정

Aperture507 committed:
fix: roleText 코드 변경

Aperture507 committed:
feat: FlatList 디자인

Aperture507 committed:
feat: Notice 디자인 변경

DonghyunKim98 commented on app/components/screens/member/Member.tsx:
```diff
@@ -10,18 +10,213 @@ import {faChevronLeft} from '@fortawesome/free-solid-svg-icons';
   const icon = (
     <FontAwesomeIcon style={{color: 'white'}} icon={faChevronLeft} />
   );
+  const renderItem: ({item}: {item: Array}) => any = ({item}) => (
```
- Array 에 대한 Generic type 이 명시되어 있지 않습니다. 명시해주세요.
- 그리고 Array 가 넘어오는게 맞나요? item 하나만 넘어오는걸로 알고있습니다.

Aperture507 replied:
<img width="631" alt="스크린샷 2021-07-23 오후 12 48 36" src="https://user-images.githubusercontent.com/81297398/126735515-b154a22c-e6b5-4e83-83f9-c6e66aa9e153.png">
Generic type에 대한 피드백은 저번에 이렇게 받긴했습니다. 린트 에러는 안뜨는데 vscode상에서는 빨간줄에 타입 명시 안되있다고 저도 뜨긴 하네요. 음...근데 저번에 이 에러 해결하려고 좀 여러가지 시도해봤는데 결국 안없어졌어서...

Aperture507 replied:
Array로 받는 것은 그냥 뭔가 item 안에 name, major, studentCode 가 있어서 Array로 당연히 받은 느낌이긴 한데, 지금 any 로 바꿔보니 빨간줄 없어지네요. any로 갈까요??

DonghyunKim98 replied:
- 반환타입을 명시 안해도 된다 라는 의미는 "함수의 반환타입"을 명시하지 않아도 된다 라는 의미였습니다.
- 즉 아래와 같이 코드를 고치는게 어떻겠냐 라는 의미였죠
```
const renderItem = ({item} : {item : Array<???>}) => {
 //rest of codes go here...
}
```

Aperture507 replied:
넵 근데 저 <>안에 하나밖에 안들어가잖아요? 근데 item 값은 string도 있고 number도 있고 그래서 뭔가 생각을 그만뒀었어요 ㅋㅋ
따로 분리해야되나

DonghyunKim98 replied:
- name, major, studentCode 등을 담은 interface 를 만들면 될것같습니다 ㅎㅎ

Aperture507 replied:
크아악 그냥 array 말고 any로 가면 안되나요 ㅋㅋㅋ

DonghyunKim98 replied:
그러면 ts 를 쓰는 의미가없습니다 😭
지금 무슨 타입이 오는지 모르는게 아닌데, 명시해주세요

DonghyunKim98 commented on app/components/screens/member/Member.tsx:
```diff
@@ -10,18 +10,213 @@ import {faChevronLeft} from '@fortawesome/free-solid-svg-icons';
+  );
+  const renderSeparator = () => {
+    return <View style={styles.itemSeparator} />;
+  };
```
- Member 함수와 직접적인 관계가 없습니다. Member 함수 밖으로 빼주세요.

DonghyunKim98 commented on app/components/screens/member/Member.tsx:
```diff
@@ -10,18 +10,213 @@ import {faChevronLeft} from '@fortawesome/free-solid-svg-icons';
+      <Text style={styles.studentCode}>{studentCode}</Text>
+    </View>
+  </View>
+);
```
- Student 경우, FlatList 의 자식 요소로 쓰이는데 이럴 경우 `React.memo()` 를 이용해서 성능 최적화를 하는게 좋습니다.
- Member Component 가 업데이트 될때, FlatList 의 모든 Student 가 Virtual DOM 과 비교하면 어플이 느려지기 때문이죠.
- 새로운 파일로 분리해주세요.

DonghyunKim98 committed:
build: install pod file

Aperture507 committed:
fix: 코드 리팩토링

Aperture507 committed:
feat: Student Member에서 분리

Aperture507 committed:
feat: FlatList type 지정

Aperture507 committed:
Merge branch 'feat/notice' of https://github.com/Hanwoollim-App/Hanwoollim-managers into feat/notice

DonghyunKim98 commented on app/components/common/NoticeItemType.tsx:
```diff
@@ -0,0 +1,4 @@
+export interface NoticeItemType {
```
- `NoticeItemInterface`
로 네이밍 바꾸어주세요!

DonghyunKim98 commented on app/components/common/Student.tsx:
```diff
@@ -41,14 +42,14 @@ const styles = StyleSheet.create({
 });
 
-const Student = ({
+const Student: React.FC<StudentType> = ({
```
- React Component 는 Arrow Function 말고 function 키워드를 사용해서 나타내기로 약속했습니다~!
- `function Student()` 로 나타내주세요!

DonghyunKim98 commented on app/components/common/StudentType.tsx:
```diff
@@ -0,0 +1,5 @@
+export interface StudentType {
```
- 역시 `StudentInterface` 로 바꾸어주세요 ㅎㅎ

DonghyunKim98 commented on app/components/screens/notice/Notice.tsx:
```diff
@@ -183,7 +156,9 @@ function Notice() {
         <View style={styles.list}>
           <FlatList
             data={DATA}
-            renderItem={renderItem}
```
- 역시 item 의 타입을 명시해주세요!

DonghyunKim98 commented on app/components/screens/notice/Notice.tsx:
```diff
@@ -183,7 +156,9 @@ function Notice() {
           <FlatList
             data={DATA}
-            renderItem={renderItem}
+            renderItem={({item}) => (
```
- 이 부분입니다 item 타입 명시해주세요!

Aperture507 committed:
fix: 파일 이름 수정

DonghyunKim98 committed:
feat: 리뷰사항 반영

- flatlist 의 item 에 대한 타입 표기
- item 대신 더 직관적인 네이밍 사용
- React.memo 대신 memo 를 import 하도록 설정
- font 관련 컴포넌트 Member 함수 외부로 변경

DonghyunKim98 committed:
fix: Student Type 위치 변경

DonghyunKim98 committed:
fix: CustomHeader 관련 사항 수정

- boolean property 가 true 인 경우 명시만 진행
- 오탈자 수정

DonghyunKim98 committed:
fix: fix the lint and the file naming

- react 요소가 없는데 .tsx 확장자 파일들 .ts 로 변경
- BtnWithIcon 의 Props 에서 icon 관련 property들 선택사항으로 변경

DonghyunKim98 committed:
fix: Notice 관련 type 표기

DonghyunKim98 committed:
Merge branch 'dev' into feat/notice
